### PR TITLE
[CONFORMANCE] Remove deprecated subgraphs dumper tool

### PR DESCRIPTION
**Pull request by iefode**
### Details:
 - *Remove old sumbgraphs dumper tool*
 - @andrei-kochin Could you please approve the PR?

### Tickets:
 - *[ticket-id](https://jira.devtools.intel.com/browse/CVS-116521)*

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: src/tests/functional/plugin/conformance/subgraphs_dumper/CMakeLists.txt [hunk 1]
```diff
@@ -2,47 +2,21 @@
...
-
-if(TARGET openvino_pytorch_frontend)
-    list(APPEND DEPENDENCIES openvino_pytorch_frontend)
-    list(APPEND DEFINITIONS ENABLE_OV_PYTORCH_FRONTEND)
-endif()
-
-if(TARGET openvino_tensorflow_frontend)
-    list(APPEND DEPENDENCIES openvino_tensorflow_frontend)
-    list(APPEND DEFINITIONS ENABLE_OV_TF_FRONTEND)
-endif()
```
[the remark this answers is not in the document]
Comment (iefode) [reply src/tests/functional/plugin/conformance/subgraphs_dumper/CMakeLists.txt:39]: of course yes

[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply src/tests/functional/plugin/conformance/subgraphs_dumper/CMakeLists.txt:39]: but I don't see it here.

File: src/tests/functional/plugin/conformance/subgraphs_dumper/CMakeLists.txt [hunk 2]
```diff
@@ -51,5 +25,4 @@ addIeTargetTest(
         ADD_CPPLINT
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/conformance/subgraphs_dumper/CMakeLists.txt:25]: `DEPENDENCIES` is always empty here. I suppose we need to add `ov_frontends` here

Comment (ilya-lavrenov) [reply src/tests/functional/plugin/conformance/subgraphs_dumper/CMakeLists.txt:47]: should we just depend on `ov_frontends` target which already depends on all frontends? It's more scalable way

**Pull request merged**